### ISSUE TITLE
🚀 Add option to combine left-right line number in traditional view

### ISSUE THREAD

**Issue by kevinhwang91** [dandavison/delta#597]
I try https://github.com/banga/git-split-diffs for a few days, and the style is extremely awesome, but the performance is slightly bad.
IMO, the two line number column is so redundant.

Sscreenshot for git-split-diffs:
![image](https://user-images.githubusercontent.com/17562139/117628956-42509280-b1ac-11eb-8c52-f8614899d05e.png)

BTW, the file-decoration-style of git-split-diffs is nice!

**Comment by dandavison** [comment]
Thanks @kevinhwang91, git-split-diffs looks really nice. Here's a comparison against delta's improved side-by-side mode (with line wrapping) that @th1000s has written. From this quick comparison it looks like they have some nice styling ideas, and that their word-wrapping algorithm can currently be a bit misleading as to what's a wrapping artifact. 

**delta**
<table><tr><td><img width=800px src="https://user-images.githubusercontent.com/52205/117661451-63918e80-b16c-11eb-8821-530883dd5e7f.png" alt="image" /></td></tr></table>

**git-split-diffs**
<table><tr><td><img width=800px src="https://user-images.githubusercontent.com/52205/117661499-773cf500-b16c-11eb-9124-54e7aa2cb290.png" alt="image" /></td></tr></table>


(delta using my current go-to light-theme settings:

```gitconfig
[delta]
    features = hoopoe
    commit-decoration-style = blue box
```
)

**Comment by kevinhwang91** [comment]
Thanks for your reply. I know the algorithm of git-split-diffs is not good enough compare with delta now, and the performance will never catch up with delta either. Although I don't use it now, I admit the neat and beautiful style it has.
I have stared either git-split-diffs or delta.

I know that combine two line number to one will produce misleading line number, but most of the times for me, I am more focus on what the difference between code.

In my workflow, I use side-by-side mode in nvim's diff mode because I must make sure the code is ready to commit and navigate code with LSP. I rarely use side-by-side in a terminal with tmux split a window into more than two panes, which is not enough width for side-by-side mode.

I think you can consider my request for the topic.

**Comment by kevinhwang91** [comment]
No, you got me wrong. I ask to compare the code up and down in the traditional view.

**Comment by dandavison** [comment]
> No, you got me wrong. I ask to compare the code up and down in the traditional view.

I'm sorry, but I don't quite understand what you mean there. Can you describe in a bit more detail or provide a picture/diagram?

**Comment by kevinhwang91** [comment]
![image](https://user-images.githubusercontent.com/17562139/117689631-89f70e80-b1ec-11eb-8974-35a5bfcd3afc.png)
top pane is delta, bottom is git-split-diffs. I ask to combine two line number into one.

**Comment by dandavison** [comment]
Ah, I see now, thanks for explaining!

Those numbers look rather confusing to me though -- why are the grey numbers from the old version rather than the new? And with the way that the numbers decrease as well as increase I think it would be hard for a human brain to properly interpret them. I wonder

> I know that combine two line number to one will produce misleading line number, but most of the times for me, I am more focused on what's the difference between code.

Do you actually need the line numbers in the old version? Or would something like this serve your purposes:

```gitconfig
[delta]
    line-numbers-left-format = ""
    line-numbers-right-format = "{np:^4}│"
```

<table><tr><td><img width=600px src="https://user-images.githubusercontent.com/52205/117715146-e2ef8400-b1a5-11eb-8430-b15dca562e26.png" alt="image" /></td></tr></table>

**Comment by TwoFinger** [comment]
> why are the grey numbers from the old version rather than the new? @dandavison

IMO, this is plain wrong. One would expect that on a "plus line" both the line number and the content should be from the "plus file". (Same for the "minus file".)

> I know that combine two line number to one will produce misleading line number, but most of the times for me, I am more focused on what's the difference between code. @kevinhwang91 

Why request a feature if you will ignore it?

> In my workflow, I use nvim's diff mode because I must make sure the code is ready to commit and navigate code with LSP. @kevinhwang91 

The feature you are proposing will not bring delta any closer to this workflow.

**Comment by kevinhwang91** [comment]
> Ah, I see now, thanks for explaining!
> 
> Those numbers look rather confusing to me though -- why are the grey numbers from the old version rather than the new? And with the way that the numbers decrease as well as increase I think it would be hard for a human brain to properly interpret them. I wonder
> 
> > I know that combine two line number to one will produce misleading line number, but most of the times for me, I am more focused on what's the difference between code.
> 
> Do you actually need the line numbers in the old version? Or would something like this serve your purposes:
> 
> ```gitconfig
> [delta]
>     line-numbers-left-format = ""
>     line-numbers-right-format = "{np:^4}│"
> ```
> 
> <img alt="image" width="600px" src="https://user-images.githubusercontent.com/52205/117715146-e2ef8400-b1a5-11eb-8430-b15dca562e26.png">

This exactly what I want.
